go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -160,48 +170,145 @@ func (oc *Controller) updateEgressFirewallWithRetry(egressfirewall *egressfirewa
+			// should not happen because this value is already validated
+			return fmt.Errorf("cannot add logical_router_policy  %s is not a valid CIDR", rule.to.cidrSelector)
 		}
-		if !utilnet.IsIPv6(ipAddress) {
-			match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s", rule.to.cidrSelector, hashedAddressSetNameIPv4)
+		if utilnet.IsIPv6(ip) {
+			matchTargets = []matchTarget{{matchKindV6CIDR, rule.to.cidrSelector}}
 		} else {
-			match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s", rule.to.cidrSelector, hashedAddressSetNameIPv6)
+			matchTargets = []matchTarget{{matchKindV4CIDR, rule.to.cidrSelector}}
 		}
+		if rule.to.cidrSelector != "" {
+			if utilnet.IsIPv6CIDRString(rule.to.cidrSelector) {
+				matchTargets = []matchTarget{{matchKindV6CIDR, rule.to.cidrSelector}}
+			} else {
+				matchTargets = []matchTarget{{matchKindV4CIDR, rule.to.cidrSelector}}
 
-		if len(rule.ports) > 0 {
-			match = fmt.Sprintf("%s && ( %s )", match, egressGetL4Match(rule.ports))
+			}
+		} else {
+			// rule based on DNS NAME
+			dnsNameAddressSets, err := oc.egressFirewallDNS.Add(ef.namespace, rule.to.dnsName)
+			if err != nil {
+				return fmt.Errorf("error with EgressFirewallDNS - %v", err)
+			}
+			if dnsNameAddressSets.GetIPv4HashName() != "" {
+				matchTargets = append(matchTargets, matchTarget{matchKindV4AddressSet, dnsNameAddressSets.GetIPv4HashName()})
+			}
+			if dnsNameAddressSets.GetIPv6HashName() != "" {
+				matchTargets = append(matchTargets, matchTarget{matchKindV6AddressSet, dnsNameAddressSets.GetIPv6HashName()})
+			}
 		}
-
-		match = fmt.Sprintf("%s && %s\"", match, getClusterSubnetsExclusion())
-
-		_, stderr, err := util.RunOVNNbctl("--id=@logical_router_policy", "create", "logical_router_policy",
-			fmt.Sprintf("priority=%d", efStartPriority-rule.id),
-			match, "action="+action, fmt.Sprintf("external-ids:egressFirewall=%s", ef.namespace),
-			"--", "add", "logical_router", ovnClusterRouter, "policies", "@logical_router_policy")
+		match := generateMatch(hashedAddressSetNameIPv4, hashedAddressSetNameIPv6, matchTargets, rule.ports)
+		err = createLogicalRouterPolicy(efStartPriority-rule.id, match, action, ef.namespace)
 		if err != nil {
-			// TODO: lr-policy-add doesn't support --may-exist, resort to this workaround for now.
-			// Have raised an issue against ovn repository (https://github.com/ovn-org/ovn/issues/49)
-			if !strings.Contains(stderr, "already existed") {
-				return fmt.Errorf("failed to add policy route '%s' to %s "+
-					"stderr: %s, error: %v", match, ovnClusterRouter, stderr, err)
-			}
+			return err
 		}
 	}
 	return nil
 }
 
+// createLogicalRouterPolicy uses the previously generated elements and creates the logical_router_policy
+// for a specific egressFirewallRouter
+func createLogicalRouterPolicy(priority int, match, action, namespace string) error {
+	_, stderr, err := util.RunOVNNbctl("--id=@logical_router_policy", "create", "logical_router_policy",
```

Comment:
all the PRs were merged - I think this can have `--may-exist` now.

Reply:
(ref: https://bugzilla.redhat.com/show_bug.cgi?id=1845109 - though not sure if this has made it down yet.)

Reply:
Downstreams it has (it's in `20.09.0-1`, and we are at `20.09.0-7`), upstream too it seems: https://src.fedoraproject.org/rpms/ovn/blob/f32/f/ovn.spec#_485